### PR TITLE
MGMT-3811 KubeAPI Minimal ISO support

### PR DESCRIPTION
**Pull request by nmagnezi**
Adds ImageType Config to InstallEnvReconciler.
Same as in done in the operator, the config will read env variable `ISO_IMAGE_TYPE` which can
either be: `minimal-iso` (default) or `full-iso`

### CODE REVIEW

File: internal/controller/controllers/installenv_controller.go [hunk 1]
```diff
@@ -153,10 +153,16 @@ func (r *InstallEnvReconciler) ensureISO(ctx context.Context, installEnv *adiiov
 		return ctrl.Result{Requeue: false}, nil
 	}
 
-	// Generate ISO
+	// Defaults to minimal-iso
+	imageType := models.ImageTypeMinimalIso
```
Comment (nmagnezi) [inline internal/controller/controllers/installenv_controller.go:157]: should be env var

File: internal/controller/api/v1alpha1/installenv_types.go [hunk 2]
```diff
@@ -45,6 +46,10 @@ const (
 )
 
 type InstallEnvSpec struct {
+	// ImageType is the type of image that should be generated.
+	// ImageType defaults to "minimal-iso".
+	ImageType models.ImageType `json:"imageType,omitempty"`
```
Comment (filanov) [inline internal/controller/api/v1alpha1/installenv_types.go:51]: Nothing should be defined in InstallEnvs, use env variables configuraiton

**Comment by nmagnezi** [comment]
KubeAPI subsystem:

```sh
Ran 6 of 158 Specs in 64.300 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 152 Skipped
--- PASS: TestSubsystem (64.35s)
PASS
ok  	github.com/openshift/assisted-service/subsystem	65.558s
```

Also double-checked the image size with:

```yaml
apiVersion: adi.io.my.domain/v1alpha1
kind: InstallEnv
metadata:
  name: myinstallenv
spec:
  clusterRef:
    name: single-node
    namespace: assisted-installer
  agentLabelSelector:
    matchLabels:
      bla: aaa
  pullSecretRef:
    name: pull-secret
  proxy:
    httpProxy: http://11.11.11.33
    httpsProxy: http://22.22.22.55
  ignitionConfigOverride: '{"ignition": {"version": "3.1.0"}, "storage": {"files": [{"path": "/tmp/example", "contents": {"source": "data:text/plain;base64,aGVscGltdHJhcHBlZGluYXN3YWdnZXJzcGVj"}}]}}'
```

Result: 100Mb

```sh
$ kubectl -n assisted-installer describe installenvs.adi.io.my.domain myinstallenv | grep URL
        f:isoDownloadURL:
  Iso Download URL:        http://192.168.49.2:31377/api/assisted-install/v1/clusters/9bf38138-77c8-48b9-9e0c-eab80571e095/downloads/image
$ curl -O http://192.168.49.2:31377/api/assisted-install/v1/clusters/9bf38138-77c8-48b9-9e0c-eab80571e095/downloads/image
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  100M  100  100M    0     0  79.3M      0  0:00:01  0:00:01 --:--:-- 79.3M
$ ls -ltrh | grep image
-rw-rw-r--. 1 nmagnezi nmagnezi 101M Mar 14 09:03 image
```

File: subsystem/kubeapi_test.go [hunk 3]
```diff
@@ -342,6 +342,46 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}, "2m", "10s").Should(Equal(validHwInfo.SystemVendor.Manufacturer))
 	})
 
+	It("deploy clusterDeployment and installEnv and validate image generated and image type", func() {
```
Comment (filanov) [inline subsystem/kubeapi_test.go:345]: why not test it in already existing test `deploy clusterDeployment and installEnv and verify cluster updates"` ?

Comment (nmagnezi) [reply subsystem/kubeapi_test.go:345]: I thought about it. wondered if we wanted to have a "basic" test without any additional params.
I can place this in the test that you've mentioned

Comment (filanov) [reply subsystem/kubeapi_test.go]: you are just validating one parameter in the DB, you case place it in a `By` section.

File: subsystem/kubeapi_test.go [hunk 4]
```diff
@@ -389,6 +389,9 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		Expect(cluster.HTTPProxy).Should(Equal("http://192.168.1.2"))
 		Expect(cluster.HTTPSProxy).Should(Equal("http://192.168.1.3"))
 		Expect(cluster.AdditionalNtpSource).Should(ContainSubstring("192.168.1.4"))
+		Expect(cluster.ImageGenerated).Should(Equal(true))
+		// InstallEnv image type defaults to minimal-iso.
```
Comment (filanov) [inline subsystem/kubeapi_test.go:393]:
```suggestion
		By("InstallEnv image type defaults to minimal-iso.")
```

File: subsystem/kubeapi_test.go (hunk 3 above)
Comment (nmagnezi) [reply subsystem/kubeapi_test.go]: will do

File: internal/controller/controllers/installenv_controller.go [hunk 5]
```diff
@@ -44,9 +44,14 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+type InstallEnvConfig struct {
+	ImageType models.ImageType `envconfig:"ISO_IMAGE_TYPE" default:"minimal-iso"`
```
Comment (tsorya) [inline internal/controller/controllers/installenv_controller.go:48]: just a question regarding design. 
Why iso_type is os env and not part of crd with some default value?

Comment (filanov) [reply internal/controller/controllers/installenv_controller.go:48]: We don't want to enable the users to set iso type at this point.

**Review by filanov — APPROVED** [review]

File: subsystem/kubeapi_test.go (hunk 3 above)
Comment (nmagnezi) [reply subsystem/kubeapi_test.go]: updated:
```sh
[kube-api]cluster installation
  deploy clusterDeployment and installEnv and verify cluster updates
  /home/nmagnezi/go/src/github.com/assisted-service/subsystem/kubeapi_test.go:345
STEP: Validate proxy settings.
STEP: Validate additional NTP settings.
STEP: InstallEnv image type defaults to minimal-iso.
```

File: subsystem/kubeapi_test.go [hunk 6]
```diff
@@ -385,10 +385,17 @@ var _ = Describe("[kube-api]cluster installation", func() {
...
 		cluster = getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForReconcileTimeout)
-		Expect(cluster.NoProxy).Should(Equal("192.168.1.1"))
-		Expect(cluster.HTTPProxy).Should(Equal("http://192.168.1.2"))
-		Expect(cluster.HTTPSProxy).Should(Equal("http://192.168.1.3"))
+		Expect(cluster.ImageGenerated).Should(Equal(true))
+		By("Validate proxy settings.", func() {
+			Expect(cluster.NoProxy).Should(Equal("192.168.1.1"))
+			Expect(cluster.HTTPProxy).Should(Equal("http://192.168.1.2"))
+			Expect(cluster.HTTPSProxy).Should(Equal("http://192.168.1.3"))
+			Expect(cluster.AdditionalNtpSource).Should(ContainSubstring("192.168.1.4"))
```
Comment (nmagnezi) [inline subsystem/kubeapi_test.go:393]: should not be here

**Review by filanov — APPROVED** [review]

**Pull request merged**